libraries/ms-drivers/src/bts_7200_current_sense.c

```diff
@@ -35,66 +43,250 @@ static StatusCode prv_init_common(Bts7200Storage *storage) {
+  } else {
+    return prv_bts_7200_enable_pin(&storage->enable_pin_0);
+  }
+}
+
+StatusCode bts_7200_enable_output_1(Bts7200Storage *storage) {
+  if (storage->enable_pin_1.fault_in_progress) {
+    return STATUS_CODE_INTERNAL_ERROR;
+  } else {
+    return prv_bts_7200_enable_pin(&storage->enable_pin_1);
+  }
+}
+
+StatusCode bts_7200_disable_output_0(Bts7200Storage *storage) {
+  return prv_bts_7200_disable_pin(&storage->enable_pin_0);
+}
+
+StatusCode bts_7200_disable_output_1(Bts7200Storage *storage) {
+  return prv_bts_7200_disable_pin(&storage->enable_pin_1);
+}
+
+bool bts_7200_get_output_0_enabled(Bts7200Storage *storage) {
+  return prv_bts_7200_get_pin_enabled(&storage->enable_pin_0);
+}
+
+bool bts_7200_get_output_1_enabled(Bts7200Storage *storage) {
+  return prv_bts_7200_get_pin_enabled(&storage->enable_pin_1);
+}
+
 StatusCode bts_7200_get_measurement(Bts7200Storage *storage, uint16_t *meas0, uint16_t *meas1) {
   AdcChannel sense_channel = NUM_ADC_CHANNELS;
-  adc_get_channel(*storage->sense_pin, &sense_channel);
+  status_ok_or_return(adc_get_channel(*storage->sense_pin, &sense_channel));
 
-  if (storage->select_pin_type == BTS7200_SELECT_PIN_STM32) {
-    gpio_set_state(storage->select_pin_stm32, STM32_GPIO_STATE_SELECT_OUT_0);
+  if (storage->select_pin.pin_type == BTS7200_PIN_STM32) {
+    gpio_set_state(storage->select_pin.select_pin_stm32, STM32_GPIO_STATE_SELECT_OUT_0);
   } else {
-    pca9539r_gpio_set_state(storage->select_pin_pca9539r, PCA9539R_GPIO_STATE_SELECT_OUT_0);
+    pca9539r_gpio_set_state(storage->select_pin.select_pin_pca9539r,
+                            PCA9539R_GPIO_STATE_SELECT_OUT_0);
   }
-  adc_read_raw(sense_channel, meas0);
+  status_ok_or_return(adc_read_converted(sense_channel, meas0));
 
-  if (storage->select_pin_type == BTS7200_SELECT_PIN_STM32) {
-    gpio_set_state(storage->select_pin_stm32, STM32_GPIO_STATE_SELECT_OUT_1);
+  if (storage->select_pin.pin_type == BTS7200_PIN_STM32) {
+    gpio_set_state(storage->select_pin.select_pin_stm32, STM32_GPIO_STATE_SELECT_OUT_1);
   } else {
-    pca9539r_gpio_set_state(storage->select_pin_pca9539r, PCA9539R_GPIO_STATE_SELECT_OUT_1);
+    pca9539r_gpio_set_state(storage->select_pin.select_pin_pca9539r,
+                            PCA9539R_GPIO_STATE_SELECT_OUT_1);
+  }
+  status_ok_or_return(adc_read_converted(sense_channel, meas1));
+  // Set equal to 0 if below/equal to leakage current.  Otherwise, convert to true load current.
+  // Check for faults, call callback and handle fault if faulted
+  bool fault0 = BTS7200_IS_MEASUREMENT_FAULT_MV(*meas0);
+  bool fault1 = BTS7200_IS_MEASUREMENT_FAULT_MV(*meas1);
+
+  if (*meas0 <= BTS7200_MAX_LEAKAGE_VOLTAGE_MV) {
+    *meas0 = 0;
+  } else {
+    (*meas0) *= BTS7200_IS_SCALING_NOMINAL;
+    (*meas0) /= storage->resistor;
+  }
+  if (*meas1 <= BTS7200_MAX_LEAKAGE_VOLTAGE_MV) {
+    *meas1 = 0;
+  } else {
+    (*meas1) *= BTS7200_IS_SCALING_NOMINAL;
+    (*meas1) /= storage->resistor;
+  }
```

Comment:
Nit: you could factor out this into a "convert from voltage to current" function. Also you don't need the brackets in `(*meas0/1)`.